modules/ci_environment/manifests/transition_logs/s3_backups.pp

```diff
@@ -0,0 +1,46 @@
+# class ci_environment::transition_logs::s3_backups
+#
+# [*directory*]
+#   Defines the directory to backup
+#
+# [*hour*]
+#   Defines the hour to run the cron job
+#
+# [*minute*]
+#   Defines the minute to run the cron job
+#
+# [*retention*]
+#   Defines the number of full backups to retain
+#
+# [*s3_bucket*]
+#   Defines the AWS S3 bucket destination for backups
+#
+# [*s3_aws_id*]
+#   Defines the AWS access id with permission to write to 
+#   the S3 bucket
+#
+# [*s3_aws_key*]
+#   Defines the AWS access secret key for the defined aws access id
+
+class ci_environment::transition_logs::s3_backups(
+  $directory = '/srv/logs/log-1',
+  $hour = 4,
+  $minute = 15,
+  $retention = 30,
```

Comment:
We should probably talk about this next week. 

Since we are doing full backups every 15 minutes of mongo, we should tell duplicity to ONLY do full backups and forget about incrementals which have no meaning in this case. Then if we keep 30 fulls, that is only 7.5 hours worth of full backups, which should be sufficient as we can do a full restore from any of those backups.